lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3310,13 +3310,49 @@ private function dispatchOnFlushEvent()
         if ($this->evm->hasListeners(Events::onFlush)) {
             $this->evm->dispatchEvent(Events::onFlush, new OnFlushEventArgs($this->em));
         }
+        foreach ($this->entityChangeSets as $oid => $entityChangeSet) {
```

Comment:
The subscribed listeners can be fetched once the class name is known. No need to fetch a listener for each single entity.